README.md

```diff
@@ -58,9 +58,38 @@ pip install -e .["all"]
 ```
 * If that command does not work, you may try the following instead
 ````zsh
-pip install -e .\[all\]
+pip install -e ."[all]"
 ````
 
+You may wish to selectively install dependencies required for testing, developing, or building certain parts of the project.
+These options are the following:
+
+- `benchmark`: Requirements needed to run the benchmarking notebooks
+- `tutorial`: Requirements needed to run the Jupyter Notebooks with the examples
+- `test`: Requirements needed to run the test suite
+- `docs`: Requirements needed to build the documentation (see the [docs](https://github.com/xgi-org/xgi/tree/main/docs))
+- `developer`: Requirements needed to format the codebase
+- `release`: Requirements needed to release new versions of XGI
+
+```sh
+pip install xgi[option]
+pip install xgi[option1,option2]
+```
+
+## Examples
+
+### Installing requirements
+
+```bash
+$ pip install -U -r requirements/default.txt
+```
+
+### Running the tests
+
+```bash
+$ pip install -U -r requirements/default.txt
+$ pip install -U -r requirements/test.txt
+```
```

Comment:
I feel like this is precise and good but it would take a lot of space in our Readme for something that is gonna be useful to only rare users, and is probably somewhat standard pip behavior. Could we for example reduce by only listing the 6 options and link to a python page explaining this?

Reply:
I like this idea. I think I can move it to the readme in the requirements folder.

Reply:
Done.